eng/ilverify.ps1

```diff
@@ -0,0 +1,140 @@
+    }
+}
+
+# Selecting the most recent runtime (e.g. last one in the list)
+[string] $runtime = $runtimes[-1]
+Write-Host " Selected runtime:`n  $runtime"
+
+# Parse path to runtime from something like "Microsoft.NETCore.App 5.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0]" to "C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0"
+[string] $runtime_path = $runtime -replace 'Microsoft.NETCore.App (?<RuntimeVersion>.+) \[(?<RuntimePath>.+)\]', '${RuntimePath}/${RuntimeVersion}'
+Write-Host " Using the following path to runtime:`n  $runtime_path"
+
+# Check whether path exists, if it doesn't something unexpected happens and needs investigation
+if (-not (Test-Path $runtime_path)) {
+    Write-Host "Path to runtime not found. Exiting..."
+    exit 1
+}
+
+# For every artifact, every configuration and TFM, run a dotnet-ilverify with references from discovered runtime directory:
+$ilverify_output = @()
+foreach ($project in $projects.Keys) {
+    foreach ($tfm in $projects[$project]) {
+        foreach ($configuration in $configurations) {
+            $dll_path = "$artifacts_bin_path/$project/$configuration/$tfm/$project.dll"
+            if (-not (Test-Path $dll_path)) {
+                Write-Host "DLL not found: $dll_path"
+                continue
+            }
+            Write-Host "Verifying $dll_path..."
+            # If there are any errors to ignore in the array, ignore them with `-g` flag
+            $ignore_errors_string =
+                if ($ignore_errors.Length -gt 0) {
+                    $ignore_errors | ForEach-Object {
+                        "-g $_"
+                    }
+                } else { "" }
+            $ilverify_cmd = "dotnet ilverify --statistics --sanity-checks --tokens $dll_path -r '$runtime_path/*.dll' -r '$artifacts_bin_path/$project/$configuration/$tfm/FSharp.Core.dll' $ignore_errors_string"
+            Write-Host "Running ilverify command:`n $ilverify_cmd"
+            # Append output to output array
+            $ilverify_output += @(Invoke-Expression "& $ilverify_cmd" -ErrorAction SilentlyContinue)
+        }
+    }
+}
+
+# Normalize output, get rid of paths in log like
+#  [IL]: Error [StackUnexpected]: [/Users/u/code/fsharp3/artifacts/bin/FSharp.Compiler.Service/Release/net9.0/FSharp.Core.dll : Microsoft.FSharp.Collections.ArrayModule+Parallel::Choose([FSharp.Core]Microsoft.FSharp.Core.FSharpFunc`2<!!0,Microsoft.FSharp.Core.FSharpOption`1<!!1>>, !!0[])][offset 0x00000081][found Byte] Unexpected type on the stack.
+# This is a quick and dirty way to do it, but it works for now.
+$ilverify_output = $ilverify_output | ForEach-Object {
+    if ($_ -match "\[IL\]: Error \[") {
+        $parts = $_ -split " "
+        "$($parts[0]) $($parts[1]) $($parts[2]) $($parts[4..$parts.Length])"
+    } elseif ($_ -match "Error\(s\) Verifying") {
+        # do nothing
+    } else {
+        $_
+    }
+}
+
+$baseline_file = Join-Path $repo_path "eng" "ilverify.bsl"
```

Comment:
The .bsl is common, but il-verify runs per project and configuration?
Can we then tell from which source an entry comes from? e.g. difference between ns20 and net90...

ilverify_output could already be maintained per run, to allow for .bsl granularity.

Reply:
ilverify_$project_$configuration_$tfm.bsl or similar?

>

Reply:
Funny enough, paths were there, but I had to cut them, because they will be different on different machines

Reply:
Separated baselines btw